library_api/repository/csv.go

```diff
@@ -0,0 +1,52 @@
+package repository
+
+import (
+	"library_api/config"
+	"library_api/entity"
+	"log"
+	"os"
+
+	"github.com/gocarina/gocsv"
+)
+
+type csvRepository struct {
+	books []entity.Book
```

Comment:
here you don't need to save users - here you need to save a connection to the file, also play around on NOT creating new connections each time we save or retrieve books from file